core/src/main/java/org/wso2/msf4j/internal/MicroservicesRegistry.java

```diff
@@ -55,16 +59,12 @@
     public MicroservicesRegistry() {
         /* If we can find the SwaggerDefinitionService, Deploy the Swagger definition service which will return the
          Swagger definition.*/
-        try {
-            Class swaggerDefinitionServiceClass = Class.forName("org.wso2.msf4j.swagger.SwaggerDefinitionService");
-            services.add(swaggerDefinitionServiceClass.getConstructor(MicroservicesRegistry.class).newInstance(this));
-        } catch (ClassNotFoundException e) {
-            log.info("SwaggerDefinitionService can't be found in classpath.");
-        } catch (NoSuchMethodException | IllegalAccessException | InstantiationException
-                | InvocationTargetException e) {
-            if (log.isDebugEnabled()) {
-                log.debug("Couldn't add the SwaggerDefinitionService.", e);
-            }
+        ServiceLoader<SwaggerService> swaggerServices = ServiceLoader.load(SwaggerService.class);
+        Iterator<SwaggerService> iterator = swaggerServices.iterator();
+        if (iterator.hasNext()) {
+            SwaggerService swaggerService = iterator.next();
+            swaggerService.init(this);
```

Comment:
It doesn't make sense to initialize all the Swagger services with the same MicroservicesRegistry. One Swagger service will handle one registry. The corresponding registry has to be tracked via the channel.id.